widget/popup_test.go

```diff
@@ -48,6 +50,7 @@ func TestPopUp_Move(t *testing.T) {
 
 	pos := fyne.NewPos(10, 10)
 	pop.Move(pos)
+	cache.Renderer(pop).Layout(pop.Size())
```

Comment:
This concerns me. This seems to be required because the Move function no longer lays out content to accommodate hitting the edge of the screen.
And so any code that calls PopUp.Move() may not work any more.

Reply:
The `Move` method of a widget should not lay out anything. That's the responsibility of the renderer. That's why the adjustments are now made there.

Reply:
In that case if the move could cause a need to layout then it should trigger it - that way the tests (and all other code) don't need this line added...

Reply:
There is no “all other code” and the tests are not normal code. I don't think that a `Move` should run `Layout` immediately. This is for several reasons:
1. I consider `Layout` as quite expensive
2. The `Move` of an element X is not important for the layout of X but for its parent p(X). Also `Move` is normally called from parent's renderer's `Layout` because the parent decides where the element has to be placed..
3. The `PopUp` is special because it mixes up two things: a container (an overlay with canvas size and 0,0 pos) and a content (a freely sizeable and positionable element). #707 describes a proposal to reduce or even eliminate the effects of this special behaviour.
4. Position adjustment due to outer constraints (window size) _is clearly renderer responsibility_. That's why the test that tests this adjustment has to ensure that the renderer is called. If #707 was addressed, the container and its content would be separated and the position won't be sent to container's `Move` but to a specific container function like `MoveContent` or alike. This would also make the test more clear because it would call `MoveContent` which then _might_ trigger a layout. But honestly I don't like the idea of the latter because it adds internal dependencies to the widget. This could be avoided by performing `Refresh` and call `Layout` from the renderer's `Refresh` if needed.

After writing all this I think I might add the `Refresh` mentioned in point 4 to `PopUp`. The `Layout` line then would vanish from the test again. Stay tuned :).

Reply:
Yes I think you're right. And this should get better now that #707 is landed.
Thanks